Count active users with Google Analytics

beszel opened the pull request:
Resolves #53

beszel committed:
Add popup that redundantly displays header text

beszel committed:
Add popup that redundantly displays header text

beszel committed:
Add popup that redundantly displays header text

beszel committed:
Add popup that redundantly displays header text

beszel committed:
Reproduce all DGtW header functionality in popup

The "Dismiss this header" button is not included in the popup, for
obvious reasons. The "Restore original content" button is fully working.
The message remains the same after restoring content.

beszel committed:
Reproduce all DGtW header functionality in popup

The "Dismiss this header" button is not included in the popup, for
obvious reasons. The "Restore original content" button is fully working.
The message remains the same after restoring content.

beszel committed:
Reproduce all DGtW header functionality in popup

The "Dismiss this header" button is not included in the popup, for
obvious reasons. The "Restore original content" button is fully working.
The message remains the same after restoring content.

beszel committed:
Reproduce all DGtW header functionality in popup

The "Dismiss this header" button is not included in the popup, for
obvious reasons. The "Restore original content" button is fully working.
The message remains the same after restoring content.

beszel committed:
Apply style to popup buttons

beszel committed:
Apply style to popup buttons

beszel committed:
Run full Puppeteer #101

Also fixed "Restore original content" button in popup.

beszel committed:
Run full Puppeteer #101

Also fixed "Restore original content" button in popup.

beszel committed:
Add e2e test for popup restore button

This uses a hardcoded popup URL, which should be changed.

beszel committed:
Add e2e test for popup restore button

This uses a hardcoded popup URL, which should be changed.

beszel committed:
Incomplete e2e tests for popup

beszel committed:
Incomplete e2e tests for popup

beszel committed:
Update tests to use popup

The extension ID is also static now.

beszel committed:
Update tests to use popup

The extension ID is also static now.

beszel committed:
Change "show changes" to a checkbox

With appropriate end-to-end tests

beszel committed:
Change "show changes" to a checkbox

With appropriate end-to-end tests

beszel committed:
Rewrite extension status request

When the popup requests the extension status from the content script,
it now receives a keyword telling it which message to display. It
previously received the entire message that it should display.

The Status enum can't be imported from `popup.js`, so a temporary
solution is used.

beszel committed:
Rewrite extension status request

When the popup requests the extension status from the content script,
it now receives a keyword telling it which message to display. It
previously received the entire message that it should display.

The Status enum can't be imported from `popup.js`, so a temporary
solution is used.

beszel committed:
Replace toggle buttons with checkboxes

beszel committed:
Replace toggle buttons with checkboxes

beszel committed:
Show the restore button when appropriate

beszel committed:
Show the restore button when appropriate

beszel committed:
Fix status enum

beszel committed:
Fix status enum

beszel committed:
Move popup style to `popup.css`

beszel committed:
Move popup style to `popup.css`

beszel committed:
Remove header

beszel committed:
Remove header

beszel committed:
Remove tests for excluded domain functions

Remove tests for `getExcludedDomain` and `getWhyExcluded`

beszel committed:
Remove tests for excluded domain functions

Remove tests for `getExcludedDomain` and `getWhyExcluded`

beszel committed:
Revert "Remove tests for excluded domain functions"

This reverts commit 2694fc24397064ea67772958dff40e5761fecd27.

The removed tests should be left in, as the functions they are testing
will be put back in. Tests will not pass on this commit.

beszel committed:
Revert "Remove tests for excluded domain functions"

This reverts commit 2694fc24397064ea67772958dff40e5761fecd27.

The removed tests should be left in, as the functions they are testing
will be put back in. Tests will not pass on this commit.

beszel committed:
Show reason that extension is disabled on a site

If the extension had to be disabled on a particular site, the header
would explain why. Now the popup explains why.

beszel committed:
Show reason that extension is disabled on a site

If the extension had to be disabled on a particular site, the header
would explain why. Now the popup explains why.

beszel committed:
Explain in popup when the content is restored

A broken test was found and fixed as part of this commit.

beszel committed:
Explain in popup when the content is restored

A broken test was found and fixed as part of this commit.

beszel committed:
Clean up IDs and selectors

beszel committed:
Clean up IDs and selectors

beszel committed:
Update version number and remove TODOs

beszel committed:
Update version number and remove TODOs

beszel committed:
Add space between status message and controls

beszel committed:
Add space between status message and controls

beszel committed:
Merge branch 'popup' of github.com:glam-lab/degender-the-web into popup

beszel committed:
Add reload button to un-restore content

beszel committed:
Make end-to-end tests more consistent and thorough

beszel committed:
Add Google Analytics for popup buttons

beszel committed:
Update privacy policy

beszel committed:
Add embedded options page with analytics checkbox

beszel committed:
Remove unused file

beszel committed:
Move analytics to background script

beszel committed:
Stop tracking reload button

This one isn't tracking for some reason.

beszel committed:
Send only a count of page views to analytics

This is used to get an accurate count of how many users actively browse
with the extension.

beszel committed:
Rename content script analytics event

beszel committed:
Update privacy policy

beszel committed:
Remove popup button analytics

We only want a user count for now.

ProfJanetDavis commented:
I see the same duplicate commits here as with #113 and I'd suggest I similar course of action. I'm sorry I got so far behind!

ProfJanetDavis commented:
Squash merge results in a reasonable set of changed files here also, though if you want a proper review through GitHub it would be helpful to have a clean PR with a minimal set of commits and changed files.

ProfJanetDavis committed:
Resolved #96. Squashed commit of the following:

commit a7f3706fca9bb459673f495b47d42478145aa00e
Author: Janet Davis <davisj@whitman.edu>
Date:   Mon Nov 25 15:20:14 2019 -0800

    Added notes from #96 to manual test cases

commit debd4ba38450d0736cdb58cc29e74df53f993351
Author: Ian Hawkins <ian@hawkapp.com>
Date:   Thu Nov 14 15:44:14 2019 -0800

    Make icon grey in more sizes

commit fbe703f868f586228af60e66d49e64a8b37bf3d1
Author: Ian Hawkins <ian@hawkapp.com>
Date:   Sat Nov 9 09:19:11 2019 -0800

    Grey out icon when it makes sense

    This doesn't work for all icon sizes.

ProfJanetDavis committed:
Merge branch 'analytics' of github.com:glam-lab/degender-the-web into analytics

beszel commented:
@ProfJanetDavis Can we close this PR, or do you want to use this Analytics experiment?

ProfJanetDavis commented:
Good question. After our last meeting, I am not sure Google Analytics is
going to serve my purposes for evaluation - I need to investigate more.
Let's leave it for now. At some point we may close the PR but leave the
branch as evidence of what you learned.

On Thu, Jan 16, 2020 at 10:08 PM Ian Hawkins <notifications@github.com>
wrote:

> Reopened #112 <https://github.com/glam-lab/degender-the-web/pull/112>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/glam-lab/degender-the-web/pull/112?email_source=notifications&email_token=AEXJIU4GYV5YYDIC6ZVLJ7TQ6FDOPA5CNFSM4JMMBQFKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOWBAHYVA#event-2957016148>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEXJIU3AMSFIFY6DVXWSIZDQ6FDOPANCNFSM4JMMBQFA>
> .
>


-- 
Janet Davis (she/her or they/them)
Associate Professor and Microsoft Chair of Computer Science, Whitman
College
509-527-5758, http://cs.whitman.edu/~davisj/
Schedule an appointment: http://meetme.so/JanetDavis
